version: "3"

services:
  traefik:
    image: traefik:v2.9
    command:
      - "--api.dashboard=true"
      - "--providers.docker=true"
      - "--entrypoints.web.address=:80"
      - "--entrypoints.websecure.address=:443"
      - "--certificatesresolvers.myresolver.acme.tlschallenge=true"
      - "--certificatesresolvers.myresolver.acme.email=bans.cowboy@gmail.com"
      - "--certificatesresolvers.myresolver.acme.storage=/letsencrypt/acme.json"
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./letsencrypt:/letsencrypt

  python:
    build: ./python-scripts
    command: bash -c "python discord_db.py && sleep 5 && python run_all.py"
    environment:
      - DATABASE_PATH=/app/db/league.db
      - STEAM_GUARD_CODE=${STEAM_GUARD_CODE}
    env_file:
      - .env.python
    volumes:
      - ./db:/app/db
    restart: "no"

  nextjs:
    build: ./nextjs-app
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.nextjs.rule=Host(`radekomsa.site`)"
      - "traefik.http.routers.nextjs.entrypoints=web,websecure"
      - "traefik.http.routers.nextjs.tls.certresolver=myresolver"
      - "traefik.http.services.nextjs.loadbalancer.server.port=3000"
    environment:
      - DATABASE_PATH=/app/db/league.db
      - NEXTAUTH_URL=https://radekomsa.site
      - AUTH_SECRET=5nfxJAUzLoY+m53Qdm/mRXFurIZi0MSDA7+uNBcR9LQ=
      - AUTH_DISCORD_ID=1341423971328852040
      - AUTH_DISCORD_SECRET=PhUIBdLwHC0LmNiQy_JNEPVyq-V5wi5x
    volumes:
      - ./db:/app/db
    depends_on:
      - python
      - traefik
    restart: always
